README.md

```diff
@@ -38,36 +71,126 @@ class GraduationsController < ApplicationController
 end
 ```
 
-Then for each request:
+Then for each request to the `index` action _HasScope_ will automatically apply the scopes as follows:
 
-```
-/graduations
-#=> acts like a normal request
+``` ruby
+# GET /graduations
+# No scopes applied
+#=> brings all graduations
+apply_scopes(Graduation).all == Graduation.all
 
-/graduations?featured=true
-#=> calls the named scope and bring featured graduations
+#GET /graduations?featured=true
+# The "featured' scope is applied
+#=> brings featured graduations
+apply_scopes(Graduation).all == Graduation.featured
 
-/graduations?by_period[started_at]=20100701&by_period[ended_at]=20101013
+# GET /graduations?by_period[started_at]=20100701&by_period[ended_at]=20101013
 #=> brings graduations in the given period
+apply_scopes(Graduation).all == Graduation.by_period('20100701', '20101013')
+
+#GET /graduations?featured=true&by_degree=phd
+#=> brings featured graduations with phd degree
+apply_scopes(Graduation).all == Graduation.featured.by_degree('phd')
 
-/graduations?featured=true&by_degree=phd
+#GET /graduations?finished=true&by_degree=phd
+#=> brings only graduations by phd degree because we didn't declared finished in our controller as a permitted scope
+
+apply_scopes(Graduation).all == Graduation.by_degree('phd')
+```
+
+### Check for applied scopes
+
+_HasScope_ creates a helper method called `current_scopes` to retrieve all the scopes applied. As it's a helper method, you'll be able to access it in the controller action or the view rendered in that action.
+
+Coming back to an example from the previous section:
+```ruby
+#GET /graduations?featured=true&by_degree=phd
 #=> brings featured graduations with phd degree
+apply_scopes(Graduation).all == Graduation.featured.by_degree('phd')
 ```
 
-You can retrieve all the scopes applied in one action with `current_scopes` method.
-In the last case, it would return: `{ featured: true, by_degree: 'phd' }`.
+Calling current_scopes would return the following:
 
-## Installation
+```
+current_scopes
+#=> { featured: true, by_degree: 'phd' }
+```
+And would be available to be called in:
+- `GraduationsController#index`
+- `views/graduations/index.html.erb` and all other partials rendered by this controller action.
 
-Add `has_scope` to your Gemfile or install it from Rubygems.
+### Usage 2: Standalone Mode
+_HasScope_ can also be used in plain-old Ruby objects (PORO). Let's refactor the previous example as a PORO used by the same controller.
+First, we'll create a bare object to house our refactor
```

Comment:
I think we can merge the two blocks below with the class & inclusion for simplification, e.g.:

```suggestion
First, we'll create a bare object and include `HasScope` to get access to its features:

```ruby
 class GraduationsSearchQuery
   include HasScope
   #...
 end
```